catalog/registry/catalog-registry-cswinputtransformer/src/main/java/ddf/catalog/registry/transformer/RegistryServiceConverter.java

```diff
@@ -0,0 +1,441 @@
+        while (reader.hasMoreChildren()) {
+            reader.moveDown();
+
+            switch (reader.getNodeName()) {
+            case "rim:Value":
+                l.add(reader.getValue());
+                break;
+            }
+
+            reader.moveUp();
+        }
+
+        reader.moveUp();
+
+        return l;
+    }
+
+    /**
+     * Read name, telephone number, and email from point of contact to attach to metacard.
+     *
+     * @param reader
+     * @param meta
+     */
+    private void readPointOfContact(HierarchicalStreamReader reader, MetacardImpl meta) {
+        reader.moveDown();
+        reader.moveDown();
+        reader.moveDown();
+
+        Map<String, String> map = new HashMap<>();
+
+        map.put("name", "no name");
+        map.put("telephone", "no telephone number");
+        map.put("email", "no email address");
+
+        while (reader.hasMoreChildren()) {
+            reader.moveDown();
+
+            switch (reader.getNodeName()) {
+            case "rim:PersonName":
+                map.put("name",
+                        reader.getAttribute("firstName") + " " + reader.getAttribute("lastName"));
+                break;
+            case "rim:TelephoneNumber":
+                map.put("telephone", readPhoneNumber(reader));
+                break;
+            case "rim:EmailAddress":
+                map.put("email", reader.getAttribute("address"));
+                break;
+            }
+
+            reader.moveUp();
+        }
+
+        String contact = String
+                .format("%s, %s, %s", map.get("name"), map.get("telephone"), map.get("email"));
+
+        meta.setAttribute(Metacard.POINT_OF_CONTACT, contact);
+
+        reader.moveUp();
+        reader.moveUp();
+        reader.moveUp();
+    }
+
+    /**
+     * Quickly validate that a service binding has:
+     * - a bindingType slot
+     *
+     * @param reader
+     * @throws ConversionException
+     */
+    private String validateServiceBinding(HierarchicalStreamReader reader)
+            throws ConversionException {
+
+        String id = reader.getAttribute("id");
+        String bindingType = null;
+
+        int bindingTypes = 0;
+        while (reader.hasMoreChildren()) {
+            reader.moveDown();
+
+            if ("bindingType".equals(reader.getAttribute("name"))) {
+
+                List<String> types = readValueList(reader);
+
+                if (types.size() != 1) {
+                    throw new ConversionException(
+                            "ServiceBinding with id = " + id + " has invalid bindingType value.");
+                }
+
+                bindingType = types.get(0);
+                bindingTypes += 1;
+            }
+
+            reader.moveUp();
+        }
+
+        if (bindingTypes == 0) {
+            throw new ConversionException(
+                    "ServiceBinding with id = " + id + " has no bindingType.");
+        } else if (bindingTypes > 1) {
```

Comment:
I would move the throwing of an error for `bindingTypes > 1` into the while loop, so that as soon as you have more than 1 bindingType the error is thrown.  

This keeps the method from doing unnecessary work, and puts the error closer to where it has actually occurred.